src/DispatchMiddleware.php

```diff
@@ -7,56 +7,10 @@
 
 namespace Zend\Expressive\Router;
 
-use Interop\Http\Middleware\ServerMiddlewareInterface as LegacyLegacyMiddlewareInterface;
-use Interop\Http\ServerMiddleware\MiddlewareInterface as LegacyMiddlewareInterface;
-use Interop\Http\Server\MiddlewareInterface as InteropMiddlewareInterface;
-use Psr\Http\Message\ResponseInterface;
-use Psr\Http\Message\ServerRequestInterface;
-use Webimpress\HttpMiddlewareCompatibility\HandlerInterface;
-use Webimpress\HttpMiddlewareCompatibility\MiddlewareInterface;
-
-use const Webimpress\HttpMiddlewareCompatibility\HANDLER_METHOD;
-
 /**
- * Default dispatch middleware.
- *
- * Checks for a composed route result in the request. If none is provided,
- * delegates to the next middleware.
- *
- * Otherwise, it pulls the middleware from the route result. If the middleware
- * is not http-interop middleware, it raises an exception. This means that
- * this middleware is incompatible with routes that store non-http-interop
- * middleware instances! Make certain you only provide middleware instances
- * to your router when using this middleware.
+ * @deprecated since 2.4.0; to remove in 3.0.0. Use
```

Comment:
I've written about it on slack, but I will note it also here, maybe we should also trigger_error `E_USER_DEPRECATED`, as it is done for example here:

https://github.com/symfony/security/blob/2.7/Core/SecurityContext.php#L14

Reply:
These middleware in the root of the package have never been integrated with a release of zend-expressive or the skeleton; the 2.2 release of those packages will use the canonical versions in the `Middleware` subnamespace. These are kept only because we released them with the 2.3 release; we likely should have waited on that release until we had a full migration plan ready.

Add to this the fact that we never documented their existence, and the number of people affected is probably zero.

As such, I don't think we need to do anything more than provide the annotations.